src/main/java/edu/ucdavis/dss/ipa/entities/InstructorType.java

```diff
@@ -0,0 +1,88 @@
+package edu.ucdavis.dss.ipa.entities;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import edu.ucdavis.dss.ipa.api.deserializers.InstructorTypeDeserializer;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@SuppressWarnings("serial")
+@Entity
+@Table(name = "InstructorTypes")
+@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
+@JsonDeserialize(using = InstructorTypeDeserializer.class)
+public class InstructorType extends BaseEntity {
+    private long id;
+    private Budget budget;
+    private Float cost;
+    private String description;
+    private List<InstructorCost> instructorCosts = new ArrayList<>();
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "Id", unique = true, nullable = false)
+    @JsonProperty
+    public long getId()
+    {
+        return this.id;
+    }
+
+    public void setId(long id)
+    {
+        this.id = id;
+    }
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "BudgetId", nullable = false)
+    @NotNull
+    @JsonIgnore
+    public Budget getBudget() {
+        return budget;
+    }
+
+    public void setBudget(Budget budget) {
+        this.budget = budget;
+    }
+
+    public Float getCost() {
+        return cost;
+    }
+
+    public void setCost(Float cost) {
+        this.cost = cost;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public void setDescription(String description) {
+        this.description = description;
+    }
+
+    @OneToMany(fetch = FetchType.LAZY, mappedBy = "instructorType", cascade = {CascadeType.ALL})
+    @JsonIgnore
+    public List<InstructorCost> getInstructorCosts() {
+        return instructorCosts;
+    }
+
+    public void setInstructorCosts(List<InstructorCost> instructorCosts) {
+        this.instructorCosts = instructorCosts;
+    }
+
+    @JsonProperty("budgetId")
+    @Transient
+    public long getBudgetIdentification() {
+        if(budget != null) {
+            return budget.getId();
+        } else {
+            return 0;
```

Comment:
Why not use 'Long' instead of 'long' and return null? Is '0' not an acceptable value?